hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1311,159 +1308,6 @@ private PersistedUserVolumeInfo getVolumesByUser(String userNameKey)
     }
   }
 
-  /**
-   * Returns a list of pending deletion key info up to the limit.
-   * Each entry is a {@link BlockGroup}, which contains the info about the key
-   * name and all its associated block IDs.
-   *
-   * @param keyCount max number of keys to return.
-   * @param omSnapshotManager SnapshotManager
-   * @return a list of {@link BlockGroup} represent keys and blocks.
-   * @throws IOException
-   */
-  public PendingKeysDeletion getPendingDeletionKeys(final int keyCount,
```

Comment:
this method is similar but not entirely the same as the one in KeyManagerImpl.
It's also super complex and hard to understand.

Reply:
yeah I moved the function to KeyManagerImpl. Most of the logic is going to be performed in the ReclaimableKeyFilter class